### DISCUSSION TITLE
Mirth-SSL connection without SSL manager

### DISCUSSION THREAD

**Discussion by kunalgupta15** [nextgenhealthcare/connect#4631] [Q&A] [answered]
I am trying to create a connection using HTTPS to sent messages through API but SSL certification on the mirth is showing with X mark. I am not using SSL manager as I am using the free mirth version. Has anyone configuration mirth SSL connectivity using HTTP sender connector on mirth without SSL manager?

**Comment by pacmano1** [comment]
Please visit the mirth forums site.  Your question and workarounds have been discussed many times.  Those workarounds involve using tools like stunnel and haproxy as proxies for the connection.  Other suggestions are there also. I'd suggest licensing the mirth core pack though which is inexpensive and immediately addresses the problem.

**Reply by rjkroll** [reply]
Yes, @pacmano1 gives the solution, which is the look for the solution on the official board.    STunnel is probably the "easiest" solution without having the official SSL plugin.    You can always build a SSL sender yourself using Java, but its complex.

**Reply by kunalgupta15** [reply]
Thanks for the response. Does it mean we can't configure SSL on open source mirth without using proxy or creating SSL sender? 

I was going through this page https://forums.mirthproject.io/forum/mirth-connect/support/17134- and it confused me that even if the SSL is configured, open source mirth will still show that "X"?

**Comment by pacmano1** [comment]
I'd reiterate for 2k a year, the core pack, which also includes the channel history plugin and a few others is a steal compared to licensing other integration engines.  And arguably a great time saver when you get more complex SSL situations like mutual TLS, http listeners, web services listeners and so on.    I don't work for Nexten.    You waste dozens of hours figuring out that crap yourself with third party tools or just click a few times.   That comes from experience with nearly all ways to do TLS.    Relative value of your and our clients' time I say.

**Reply by kunalgupta15** [reply]
Thanks for the response. I am checking with Mirth sales team about the pricing. I remember few years back they use to have pricing available on their website but now I can see that.

**Reply by 5011-5011** [reply]
@pacmano1 Hello! Check the content where you indicate how to configure stunnel. I understand that it must be on Linux, are there any more specifications? Thanks

**Reply by pacmano1** [reply]
I don't understand what you are asking.  And stunnel runs on windows also.  The documentaion for stunnel is robust at https://www.stunnel.org/.  And I wrote about windows at https://forums.mirthproject.io/forum/mirth-connect/support/176520-stunnel-for-https-receiver?p=176533#post176533

**Reply by pacmano1** [reply]
Also see https://github.com/pacmano1/mirthstunnel.  While I wrote it and it works, I do need to clean it up one of these days.

**Comment by jonbartels** [comment, marked as answer]
https://gist.github.com/jonbartels/8abd121901eb930f46245d9ef0f5710e describes the different options for SSL with MC. It includes a specific line item answering your question about the HTTP sender.
